clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -374,19 +376,37 @@ public Subscription subscription(Set<String> topics) {
         if (memberAssignment == null)
             return new Subscription(new ArrayList<>(topics));
 
-        return new Subscription(new ArrayList<>(topics),
-                serializeTopicPartitionAssignment(new ConsumerUserData(memberAssignment, Optional.of(generation))));
+        return buildSubscription(topics);
     }
 
     @Override
     public String name() {
         return "sticky";
     }
 
+    List<TopicPartition> memberAssignment() {
+        return memberAssignment;
+    }
+
     int generation() {
         return generation;
     }
 
+    // This method may be overriden by a subclass that follows sticky assignment but encodes Subscription/UserData differently
```

Comment:
What's the purpose of this comment?

Reply:
A note to myself while I think about further refactoring (possibly extract into abstract class?)